adal/src/main/java/com/microsoft/aad/adal/TokenCacheAccessor.java

```diff
@@ -306,11 +304,17 @@ void updateCachedItemWithResult(final AuthenticationRequest request, final Authe
     /**
      * Update token cache with returned auth result.
      */
-    void updateTokenCache(final AuthenticationRequest request,  final AuthenticationResult result) throws MalformedURLException {
+    void updateTokenCache(final AuthenticationRequest request, final AuthenticationResult result) throws MalformedURLException {
         if (result == null || StringExtensions.isNullOrBlank(result.getAccessToken())) {
             return;
         }
 
+        if (!StringExtensions.isNullOrBlank(result.getAuthority())
```

Comment:
Should there also be a test update?

Reply:
+1

Reply:
Not so much an "update" because no tests exist for this class _at all_.

However, I've taken the liberty of adding a test in 4c76469f479696f115ac4d734c8a2790cd417d54